aws-lc-rs/src/ptr.rs

```diff
@@ -85,7 +88,7 @@ impl<P: Pointer> DetachableLcPtr<P> {
     }
 }
 
-impl<P: Pointer + Copy> DetachableLcPtr<P> {
+impl<P: Pointer + Copy> DetachablePointer<P> {
     #[inline]
     pub fn as_const<T>(&self) -> ConstPointer<T> {
```

Comment:
This has been bothering me -- `T` should be an associated type for the Pointer instead of a generic for this function.